metal3-templater.go

```diff
@@ -64,6 +70,14 @@ func main() {
 	templateData.ClusterIronicURL = fmt.Sprintf("http://%s", net.JoinHostPort(ip, "6385"))
 	templateData.ClusterInspectorURL = fmt.Sprintf("http://%s", net.JoinHostPort(ip, "5050"))
 
+	// http_basic credentials
+	if numArgs == 11 {
```

Comment:
Counting numbers of arguments doesn't really seem like a sustainable approach to extending tools like this. We need to be able to specify what the arguments mean. This approach does not give us the flexibility to change to a different auth method (upstream is probably going to implement mTLS auth at some point). The [flag](https://golang.org/pkg/flag/) package in the standard library looks like it provides a relatively simple way to implement proper CLI parsing.

Reply:
I've update to use flag, going to do some tests and update the PR